src/stores/wpcom-api.ts

```diff
@@ -0,0 +1,70 @@
+import { createApi } from '@reduxjs/toolkit/query/react';
+import { z } from 'zod';
+import { useOffline } from 'src/hooks/use-offline';
+import type { BaseQueryFn, FetchBaseQueryError } from '@reduxjs/toolkit/query';
+import type WPCOM from 'wpcom';
+
+const welcomeMessageSchema = z.object( {
+	messages: z.array( z.string() ),
+	example_prompts: z.array( z.string() ),
+} );
+
+export type WelcomeMessageResponse = z.infer< typeof welcomeMessageSchema >;
+
+let wpcomClient: WPCOM | undefined;
+
+export const setWpcomClient = ( client: WPCOM | undefined ) => {
+	wpcomClient = client;
+};
+
+const wpcomBaseQuery: BaseQueryFn<
+	{ path: string; apiNamespace?: string },
+	unknown,
+	FetchBaseQueryError
+> = async ( args ) => {
+	try {
+		const response = await wpcomClient!.req.get( args );
```

Comment:
```suggestion
> = async ( args, api ) => {
	const state = api.getState();
	const wpcomClient = state.auth.wpcomClient;

	try {
		const response = await wpcomClient!.req.get( args );
```
We don't have to make this change in this PR, but I realize that we should store `wpcomClient` in the Redux state to make it accessible in other slices/thunks, too.

Reply:
I started to think about that, but I believe the general best practice is to not store complex objects in the Redux store.
I think there are some other alternatives we could explore, like middlewares, but it seems complex enough to warrant its own PR.